processing/src/main/java/org/apache/druid/query/ordering/StringComparators.java

```diff
@@ -416,4 +420,50 @@ public boolean equals(Object o)
       return new byte[]{(byte) NUMERIC_CACHE_ID};
     }
   }
+
+  public static class VersionComparator extends StringComparator
+  {
+    @Override
+    public int compare(String o1, String o2)
+    {
+      if (o1 == o2) {
```

Comment:
teamcity check wants this to be `.equals` instead of `==`, tthough in `LexicographicComparator.compare` i see a comment about assuming most strings being compared are different and so this is a reference equality check. 

```
// Avoid conversion to bytes for equal references
// Assuming we mostly compare different strings, checking s.equals(s2) will only make the comparison slower.
```

I don't know if that is as true with this comparator, where i would expect a smaller set of strings, but if that makes sense to do here, add 
```
//noinspection StringEquality
```

Reply:
Yes, I think the number of version is small, I'll add   `//noinspection StringEquality` here, thanks su much!